packages/types/src/Method.ts

```diff
@@ -80,21 +80,29 @@ export default class Method extends Struct {
       value.methodIndex &&
       value.args
     ) {
+      // destructure value, we only pass args/methodIndex out
+      const { args, methodIndex } = value;
+
       // Get the correct callIndex
-      const callIndex = value.methodIndex instanceof MethodIndex
-        ? value.methodIndex.callIndex
-        : value.methodIndex;
+      const callIndex = methodIndex instanceof MethodIndex
```

Comment:
Not for this PR. But I'm really wondering if Method should not just get rid of MethodIndex and just encapsulate the callIndex value in here. The whole methodIndex/callIndex dance is just getting cumbersome and adds some complexity and confusion. (It really is never seperated from Method itself, so it is overkill)